agent/agent.go

```diff
@@ -518,6 +500,15 @@ func (a *Agent) setupTask(agentCtx, setupCtx context.Context, tcInput *taskConte
 	return tc, setupFailed, shouldExit, nil
 }
 
+func (a *Agent) handleSetupError(err error, tc *taskContext, ctx context.Context, shouldSetLogger bool) (*taskContext, bool, bool, error) {
```

Comment:
nit: I would call `shouldSetLogger` something more meaningful like `shouldUseFallbackLogger` (i.e. the logger when we can't set up the logger). Alternatively, initializing and passing in the logger in the error cases where the logger is not set up might be cleaner than a bool.

Reply:
I moved `tc.logger = client.NewSingleChannelLogHarness("agent.error", a.defaultLogger)` out so that bool is not needed.